R/db_download.R

```diff
@@ -62,9 +63,13 @@ db_download_ncbi <- function(verbose = TRUE, overwrite = FALSE) {
 
   # make home dir if not already present
   tdb_cache$mkdir()
-  # download data
-  mssg(verbose, 'downloading...')
-  curl::curl_download(db_url, db_path_file, quiet = TRUE)
+  if(is.null(path)){
+    # download data
+    mssg(verbose, 'downloading...')
+    curl::curl_download(db_url, db_path_file, quiet = TRUE)
+  } else {
+    file.copy(path, db_path_file)
```

Comment:
If there is already a `taxdump.zip` in the cache directory, then `file.copy()` will not do anything because the default behaviour of the function is `overwrite = FALSE`. However if we call `db_download_ncbi()` with `overwrite = TRUE` and a manually downloaded `taxdump.zip`, we might expect to overwrite the old cache file with the new, manually downloaded file. Maybe the `overwrite` argument should be called explicitly?

```
file.copy(path, db_path_file, overwrite = overwrite)
``` 

This way if `db_download_ncbi()` is called with `overwrite = TRUE` then the manually downloaded `taxdump.zip` will overwrite matching files in the cache directory.